src/main/java/com/kapoorlabs/kiara/domain/SdqlNode.java

```diff
@@ -56,26 +57,11 @@ public boolean equals(Object obj) {
 		if (!(obj instanceof SdqlNode))
 			return false;
 		SdqlNode other = (SdqlNode) obj;
-		if (doubleValue == null) {
-			if (other.doubleValue != null)
-				return false;
-		} else if (!doubleValue.equals(other.doubleValue))
-			return false;
-		if (lowerBound != other.lowerBound)
-			return false;
-		if (parent == null) {
-			if (other.parent != null)
-				return false;
-		} else if (parent != other.parent)
-			return false;
-		if (stringValue == null) {
-			if (other.stringValue != null)
-				return false;
-		} else if (!stringValue.equals(other.stringValue))
-			return false;
-		if (upperBound != other.upperBound)
-			return false;
-		return true;
+		return Objects.equals(doubleValue, other.doubleValue) &&
+		       Objects.equals(lowerBound, other.lowerBound) &&
+		       Objects.equals(parent, other.parent) &&
```

Comment:
Thanks so much for making this cleaner!
All others look good, But I will double check `Objects.equals(parent, other.parent)` this weekend.

This equals will most likely trigger a recursive equal check, for example to check if parent objects are equal, it will check if grand parents are equal and so on.

I think it is sufficient to just do a memory reference check using == of parent object. so that it doesn't trigger recursive equality and impacts performance.

Reply:
Thank you for reviewing
> This equals will most likely trigger a recursive equal check, for example to check if parent objects are equal, it will check if grand parents are equal and so on.
I think it is sufficient to just do a memory reference check using == of parent object. so that it doesn't trigger recursive equality and impacts performance.

You are absolutely correct in your assessment of this.

Reply:
Addressed in 4ec1eca

I also reordered the checks, due to this reference check being of higher priority.